src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+            ResultSet resultSet = getDriversByCarIdStatement.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(parseDriver(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get drivers for car with id "
+                    + carId + " ", e);
+        }
+    }
+
+    private int removeDriversFromCar(Car car) {
+        String deleteCarsDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarsDriversStatement
+                        = connection.prepareStatement(deleteCarsDriversQuery)) {
+            deleteCarsDriversStatement.setLong(1, car.getId());
+            return deleteCarsDriversStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car drivers for " + car + " ", e);
+        }
+    }
+
+    private int addDriversToCar(Car car) {
+        int addedDrivers = 0;
+        if (car.getDrivers() == null) {
+            return addedDrivers;
+        }
+        for (Driver driver : car.getDrivers()) {
+            String addCarsDriversRelationQuery = "INSERT INTO cars_drivers (car_id, driver_id) "
+                    + "VALUES (?, ?)";
+            try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement addCarsDriversRelationStatement
+                            = connection.prepareStatement(addCarsDriversRelationQuery)) {
+                addCarsDriversRelationStatement.setLong(1, car.getId());
+                addCarsDriversRelationStatement.setLong(2, driver.getId());
+                addedDrivers += addCarsDriversRelationStatement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Couldn't add drivers for " + car + " ", e);
+            }
+        }
+        return addedDrivers;
+    }
+
+    private Car parseCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject(1, Long.class));
+        car.setModel(resultSet.getString(2));
+        car.setManufacturer(parseManufacturer(resultSet));
+        return car;
+    }
+
+    private Manufacturer parseManufacturer(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject(3, Long.class));
+        manufacturer.setName(resultSet.getString(4));
+        manufacturer.setCountry(resultSet.getString(5));
+        return manufacturer;
+    }
+
+    private Driver parseDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject(1, Long.class));
+        driver.setName(resultSet.getString(2));
+        driver.setLicenseNumber(resultSet.getString(3));
```

Comment:
Why do we use numbers of columns? If someone swaps them? It would be better to use columns' names and declare them as constants

Reply:
Constants created